_source/_posts/2022-07-21-jhipster-ci-cd.md

```diff
@@ -0,0 +1,933 @@
+            - v1-dependencies-
+      - run:
+          name: Print Java Version
+          command: 'java -version'
+      - run:
+          name: Print Node Version
+          command: 'node -v'
+      - run:
+          name: Print NPM Version
+          command: 'npm -v'
+      - run:
+          name: Install Node Modules
+          command: 'npm install'
+      - save_cache:
+          paths:
+            - node
+            - node_modules
+            - ~/.gradle
+          key: {%raw%}v1-dependencies-{{ checksum "build.gradle" }}-{{ checksum "package-lock.json" }}{%endraw%}
+      - run:
+          name: Give Executable Power
+          command: 'chmod +x gradlew'
+      - run:
+          name: Backend tests
+          command: npm run ci:backend:test
+      - run:
+          name: Build Spring Boot Docker Image
+          command: ./gradlew bootBuildImage --imageName=$IMAGE_NAME
+      - run:
+          name: Publish Docker Image
+          command: |
+            docker login -u $DOCKERHUB_USERNAME -p $DOCKERHUB_PASS
+            docker push $IMAGE_NAME
+```
+
+Do the same updates to the `gateway/.circleci/config.yml` file. Additionally, the following change is required:
+
+- Add `~/.cache/Cypress` as a cache path
+
+Be sure to replace `your-dockerhub-username` and set the correct `IMAGE_NAME`.
+The final `config.yml` for the `gateway` must look like this:
+
+```yml
+version: 2.1
+jobs:
+  build:
+    environment:
+      DOCKERHUB_USERNAME: your-dockerhub-username
+      IMAGE_NAME: $DOCKERHUB_USERNAME/gateway
+    machine:
+      image: ubuntu-2004:current
+    resource_class: large
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - {%raw%}v1-dependencies-{{ checksum "build.gradle" }}-{{ checksum "package-lock.json" }}{%endraw%}
+            # Perform a Partial Cache Restore (https://circleci.com/docs/2.0/caching/#restoring-cache)
+            - v1-dependencies-
+      - run:
+          name: Print Java Version
+          command: 'java -version'
+      - run:
+          name: Print Node Version
+          command: 'node -v'
+      - run:
+          name: Print NPM Version
+          command: 'npm -v'
+      - run:
+          name: Install Node Modules
+          command: 'npm install'
+      - save_cache:
+          paths:
+            - node
+            - node_modules
+            - ~/.gradle
+            - ~/.cache/Cypress
+          key: {%raw%}v1-dependencies-{{ checksum "build.gradle" }}-{{ checksum "package-lock.json" }}{%endraw%}
+      - run:
+          name: Give Executable Power
+          command: 'chmod +x gradlew'
+      - run:
+          name: Backend tests
+          command: npm run ci:backend:test
+      - run:
+          name: Run Front End Tests
+          command: npm run ci:frontend:test
+      - run:
+          name: Build Spring Boot Docker Image
+          command: ./gradlew bootBuildImage --imageName=$IMAGE_NAME
```

Comment:
Building the image w/o the `prod` profile causes the gateway to fail to start on Kubernetes. Since we typically recommend Jib for building JHipster, I believe you should collapse this step and the following one into a single step:

```yaml
      - run:
          name: Build and Publish Docker Image
          command: |
            ./gradlew -Pprod bootJar jib \
              -Djib.to.image=$IMAGE_NAME \
              -Djib.to.auth.username=$DOCKERHUB_USER \
              -Djib.to.auth.password=$DOCKERHUB_PASS
```

Reply:
I added the suggested step.